src/lib/config_parser.c

```diff
@@ -129,6 +129,11 @@ char *singularity_config_get_value(char *key) {
     return(NULL);
 }
 
+char *singularity_config_get_value_default(char *key, const char *def) {
```

Comment:
In that case would it make sense to leave this out of the PR for now, and then go in and give the config system a bit of a facelift to account for all of this in a separate PR? That way we don't mix and match what we're solving in each PR and what we're doing in the code, but rather go through and systematically change what needs to be changed.